src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -1542,6 +1546,118 @@ private void calculateUNF(DataFile dataFile, int varnum, Float[] dataVector) {
+        for (Long fileId : datafileIds) {
+            fixMissingOriginalType(fileId);
+        }
+    }
+    
+    // This method fixes a datatable object that's missing the format type of 
+    // the ingested original. It will check the saved original file to 
+    // determine the type. 
+    private void fixMissingOriginalType(Long fileId) {
+        DataFile dataFile = fileService.find(fileId);
+
+        if (dataFile != null && dataFile.isTabularData()) {
+            String originalFormat = dataFile.getDataTable().getOriginalFileFormat();
+            if (StringUtil.isEmpty(originalFormat) || originalFormat.equals(FileUtil.MIME_TYPE_TAB)) {
+
+                // We need to determine the mime type of the saved original
+                // and save it in the database. 
+                // 
+                // First, we need access to the file. Note that the code below 
+                // works with any supported DataFileIO driver (although, as of now
+                // all the production installations out there are only using filesystem
+                // access; but just in case)
+                // The FileUtil method that determines the type takes java.io.File 
+                // as an argument. So for DataFileIO drivers that provide local 
+                // file access, we'll just go directly to the stored file. For 
+                // swift and similar implementations, we'll read the saved aux 
+                // channel and save it as a local temp file. 
+                
+                DataFileIO dataFileIO;
+                File savedOriginalFile = null;
+                boolean tempFileRequired = false;
+                
+                try {
+                    dataFileIO = dataFile.getDataFileIO();
+                    dataFileIO.open();
+
+                    if (dataFileIO.isLocalFile()) {
+                        try {
+                            savedOriginalFile = dataFileIO.getAuxObjectAsPath(FileUtil.SAVED_ORIGINAL_FILENAME_EXTENSION).toFile();
+                        } catch (IOException ioex) {
+                            // do nothing, just make sure savedOriginalFile is still null:
+                            savedOriginalFile = null;
+                        }
+                    }
+
+                    if (savedOriginalFile == null) {
+                        tempFileRequired = true;
+
+                        ReadableByteChannel savedOriginalChannel = (ReadableByteChannel) dataFileIO.openAuxChannel(FileUtil.SAVED_ORIGINAL_FILENAME_EXTENSION);
+                        savedOriginalFile = File.createTempFile("tempSavedOriginal", ".tmp");
+                        FileChannel tempSavedOriginalChannel = new FileOutputStream(savedOriginalFile).getChannel();
+                        tempSavedOriginalChannel.transferFrom(savedOriginalChannel, 0, dataFileIO.getAuxObjectSize(FileUtil.SAVED_ORIGINAL_FILENAME_EXTENSION));
+
+                    }
+                } catch (Exception ex) {
```

Comment:
I didn't think there was a need to catch a more specific exception (the important part is that it failed, for whatever reason; regardless of what kind of exception it was, there's nothing we can do, except to log it and let the admin know what happened). But I have added an extra logging line, to leave a record of what exactly happened and what kind of exception it was (as in ex.getClass()).